examples/abitest/module/rust/src/lib.rs

```diff
@@ -713,47 +730,85 @@ impl FrontendNode {
     }
 
     fn test_backend_roundtrip(&self) -> std::io::Result<()> {
-        // Ask the backend node to transmute something.
-        let internal_req = InternalMessage {
-            msg: "aaa".to_string(),
-        };
-        let serialized_req = serde_json::to_string(&internal_req)?;
-        info!("send serialized message to backend: {}", serialized_req);
-        let mut backend_channel = oak::io::Channel::new(self.backend_out);
-        backend_channel.write_all(&serialized_req.into_bytes())?;
-
-        // Block until there is a response from the backend available.
-        let read_handles = vec![self.backend_in];
-        match oak::wait_on_channels(&read_handles) {
-            Ok(_ready_handles) => (),
-            Err(status) => {
-                return Err(std::io::Error::new(
-                    std::io::ErrorKind::Other,
-                    format!("Wait failure {:?}", status),
-                ));
-            }
+        // Make a collection of new channels for the backend nodes to read from,
+        // and send the read handles to each backend node.
+        const COUNT: usize = 3;
+        let mut read_handles = Vec::with_capacity(COUNT);
+        let mut write_handles = Vec::with_capacity(COUNT);
+        for _i in 0..COUNT {
+            let (new_write, new_read) = oak::channel_create().unwrap();
+            write_handles.push(new_write);
+            read_handles.push(new_read.handle);
+        }
+        for j in 0..BACKEND_COUNT {
+            expect_eq!(
+                OakStatus::OK,
+                oak::channel_write(self.backend_out[j], &[], &read_handles)
+            );
+        }
+        for new_read in read_handles.iter() {
+            oak::channel_close(*new_read);
         }
 
-        let mut buffer = Vec::<u8>::with_capacity(256);
-        let mut handles = Vec::with_capacity(1);
-        expect_eq!(
-            OakStatus::OK,
-            oak::channel_read(self.backend_in, &mut buffer, &mut handles)
-        );
+        // Ask the backend node to transmute something by writing a serialized
+        // request to one of the new channels that the backends just received
+        // the read handles for.
+        for new_write in write_handles.iter() {
+            let internal_req = InternalMessage {
+                msg: "aaa".to_string(),
+            };
+            let serialized_req = serde_json::to_string(&internal_req)?;
+
+            info!(
+                "send serialized message to new channel {}: {}",
+                new_write.handle, serialized_req
+            );
+            let mut new_channel = oak::io::Channel::new(*new_write);
+            new_channel.write_all(&serialized_req.into_bytes())?;
+            oak::channel_close(new_write.handle);
+
+            // Block until there is a response from one of the backends
+            // available.
+            let readies = match oak::wait_on_channels(&self.backend_in) {
```

Comment:
Something like this should work:

```suggestion
            let readies = oak::wait_on_channels(&self.backend_in)
              .map_err(|status| std::io::Error::new(
                        std::io::ErrorKind::Other,
                        format!("Wait failure {:?}", status),
                    ))?;
```

It may be necessary to implement `From<std::io::Error>` for the type returned here though, if it's not already.

Reply:
Neat, done.